.github/workflows/task-query-diff.yml

```diff
@@ -0,0 +1,61 @@
+name: Query Diff
+
+# Documentation: https://redislabs.atlassian.net/wiki/spaces/DX/pages/3967844669/RediSearch+CI+refactor
+
+on:
+  workflow_call:
+
+jobs:
+  query-diff_task:
+    name: Query Diff
+    runs-on: ${{ vars.RUNS_ON || 'ubuntu-latest' }}
+    defaults:
+      run:
+        shell: bash -l -eo pipefail {0}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup
+        working-directory: .install
+        run: ./install_script.sh sudo
+      - name: Install aws cli
+        working-directory: .install
+        run: ./install_aws.sh ${{ steps.mode.outputs.mode }}
+      - name: Setup tests dependencies
+        run: |
+          .install/test_deps/common_installations.sh
+      # TODO we might want to configure AWS credentials & use signed requests to do things the "proper" way
+      # - name: Configure AWS Credentials Using Role (node20)
+      #   if: vars.USE_AWS_ROLE == 'true' && steps.node20.outputs.supported == 'true'
+      #   uses: aws-actions/configure-aws-credentials@v4
+      #   with:
+      #     role-to-assume: ${{ vars.ARTIFACT_UPLOAD_AWS_ROLE_TO_ASSUME }}
+      #     aws-region: ${{ vars.ARTIFACT_UPLOAD_AWS_REGION }}
+      # - name: Configure AWS Credentials Using Keys (node20)
+      #   if: vars.USE_AWS_ROLE == 'false' && steps.node20.outputs.supported == 'true'
+      #   uses: aws-actions/configure-aws-credentials@v4
+      #   with:
+      #     aws-access-key-id: ${{ secrets.ARTIFACT_UPLOAD_AWS_ACCESS_KEY_ID }}
+      #     aws-secret-access-key: ${{ secrets.ARTIFACT_UPLOAD_AWS_SECRET_ACCESS_KEY }}
+      #     aws-region: ${{ vars.ARTIFACT_UPLOAD_AWS_REGION }}
+      - name: Download Baseline RediSearch
+        run: |
+          aws s3 cp s3://redismodules/redisearch/snapshots/${{ env.file }} bin/query_diff/${{ env.file }} --no-sign-request
+          unzip -d bin/query_diff bin/query_diff/${{ env.file }}
+          ls bin/query_diff
+        env:
+          file: redisearch.Linux-ubuntu24.04-x86_64.master.zip
+      - name: Build Changeset RediSearch
+        run: ./build.sh
+      - name: Run Query Diff Tests
+        run: |
+          ls ${{ env.baseline_so }}
+          ls ${{ env.changeset_so }}
+          cargo run --manifest-path src/redisearch_rs/Cargo.toml \
+            -p shadow_tester \
+            --baseline_so ${{ env.baseline_so }} \
+            --changeset_so ${{ env.changeset_so }} \
+            --rltest-path tests/pytests/runtests.sh
+        env:
+          baseline_so: bin/query_diff/module-enterprise.so
+          changeset_so: bin/linux-x64-release/search-community/redisearch.so
```

Comment:
Enterprise and search-community are not 1:1 since they use a different coordinator.
Don't we have snapshots the search community build?

Reply:
Yeah I figured as much, but I don't think there are snapshots for the non-enterprise version. Or at least non I could find with my nonexistent access to the S3 bucket 🤷🏻‍♂️ 
The only file I could reverse engineer is `s3://redismodules/redisearch/snapshots/redisearch.Linux-ubuntu24.04-x86_64.master.zip` (plus ofc for other platforms) and unzipping that only give you a .json manifest and the enterprise module